suite-common/bluetooth/src/bluetoothReducerCreator.ts

```diff
@@ -0,0 +1,127 @@
+
+export type BluetoothDevice = {
+    id: string;
+    name: string;
+    data: number[]; // Todo: consider typed data-structure for this
+    lastUpdatedTimestamp: number;
+    status: DeviceBluetoothStatus | null;
+};
+
+export type DeviceBluetoothStatusType = DeviceBluetoothStatus['type'];
+
+type BluetoothState<T extends BluetoothDevice> = {
+    adapterStatus: 'unknown' | 'enabled' | 'disabled';
+    scanStatus: BluetoothScanStatus;
+    nearbyDevices: T[];
+
+    // This will be persisted, those are devices we believed that are paired
+    // (because we already successfully paired them in the Suite) in the Operating System
+    knownDevices: T[];
+};
+
+export const bluetoothReducerCreator = <T extends BluetoothDevice>() => {
+    const initialState: BluetoothState<T> = {
+        adapterStatus: 'unknown',
+        scanStatus: 'idle',
+        knownDevices: [] as T[],
+        nearbyDevices: [] as T[],
+    };
+
+    return createReducerWithExtraDeps<BluetoothState<T>>(initialState, (builder, extra) =>
+        builder
+            .addCase(bluetoothAdapterEventAction, (state, { payload: { isPowered } }) => {
+                state.adapterStatus = isPowered ? 'enabled' : 'disabled';
+                if (!isPowered) {
+                    state.nearbyDevices = [];
+                    state.scanStatus = 'idle';
+                }
+            })
+            .addCase(
+                bluetoothNearbyDevicesUpdateAction,
+                (state, { payload: { nearbyDevices } }) => {
+                    state.nearbyDevices = nearbyDevices.sort(
+                        (a, b) => a.lastUpdatedTimestamp - b.lastUpdatedTimestamp,
+                    ) as Draft<T>[];
+                },
+            )
+            .addCase(
+                bluetoothConnectDeviceEventAction,
+                (state, { payload: { id, connectionStatus } }) => {
+                    const device = state.nearbyDevices.find(it => it.id === id);
+
+                    if (device !== undefined) {
+                        device.status = connectionStatus;
+                    }
+                },
+            )
+            .addCase(bluetoothScanStatusAction, (state, { payload: { status } }) => {
+                state.scanStatus = status;
+            })
+            .addCase(deviceActions.deviceDisconnect, (state, { payload: { bluetoothProps } }) => {
+                if (bluetoothProps) {
+                    state.nearbyDevices = state.nearbyDevices.filter(
+                        it => it.id !== bluetoothProps.id,
+                    );
+                }
+            })
+            .addCase(
+                deviceActions.connectDevice,
+                (
+                    state,
+                    {
+                        payload: {
+                            device: { bluetoothProps },
+                        },
+                    },
+                ) => {
+                    if (bluetoothProps && bluetoothProps.id in state.nearbyDevices) {
```

Comment:
This condition seems to always evaluate to `false` since `nearbyDevices` is no longer an object but an array.

```suggestion
                    if (!bluetoothProps || !bluetoothProps.id) {
                        return;
                    }
```

Reply:
:+1: